### PR TITLE
Item ID Revamp

### PR DESCRIPTION
**Pull request by ultima-soul**
## Description

Revamp item ids to be more organized and add in defines for missing items as per the agreed-upon list.

This also removes the Ball catch modifiers code that should be added via a companion PR to the Battle Engine as in-battle item effects should go there. Also, some minor revamp to Balls have been done to make them not require Master Ball as the first ball. 

Additionally, support for skipping over TM51 to TM100 for compatibility is added so vanilla compatibility is maintained and should be further iterated upon if/when we do some basic TM expansion.

Also some items id constants and item names were renamed to match newer versions. This was already done a bit for some names before so this further iterates on that. In the future, we should look to update item icons where appropriate for items that existed in Gen 3.

Feel free to suggest any items that we could add too.
## **Discord contact info**

UltimaSoul#4017

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
I accidentally pressed curly brace when pressing enter to solve the conflict. Can you please fix it? ^^;

**Comment by AsparagusEduardo** [comment]
Reminder that there we still need a companion PR for BE.

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Preliminary review: I get why you decided to delete FRLG items, but it's gonna cause conflicts with people who have already implemented uses for them, like TheXaman's TM Case implementation or his HGSS Pokédex, where he uses Teachy TV's sprite.

File: src/data/items.h [hunk 1]
```diff
@@ -1015,4303 +941,4070 @@ const struct Item gItems[] =
...
-        .name = _("Devon Scope"),
-        .itemId = ITEM_DEVON_SCOPE,
+        .name = _("Red Orb"),
+        .itemId = ITEM_RED_ORB,
         .price = 0,
-        .description = sDevonScopeDesc,
-        .importance = 1,
+        .description = sRedOrbDesc,
+        .importance = 2,
         .pocket = POCKET_KEY_ITEMS,
```
[the remark this answers is not in the document]
Comment (ultima-soul) [reply src/data/items.h:3598]: I think that once we do some old item updates (icons, descriptions, etc.) we can update items or add configs where necessary. This was just shifting around the item data for now.

**Comment by ultima-soul** [comment]
> Safari Balls disappear when being thrown

Fixed!

**Comment by ultima-soul** [comment]
Updated!

Added back Bicycle, TM Case, and Vs. Seeker items from FRLG. Also added in Poké Radar as it's also been featured in several Gens as an item with functionality.

Merged with latest IE. Removed a few unnecessary `BATTLE_ENGINE` define checks in recent commits as those changes are fine to have without ifdefs since they don't affect IE negatively, similar to other added data that's only used in other branches. 

Also removed `I_SOUL_DEW_BOOST`, as it was redundant with `B_SOUL_DEW_BOOST` (which *does* require usage of `BATTLE_ENGINE` check since it only exists in BE). This last change is used where BE code requires certain data based on a BE config. This has been done before with Mental Herb, where its item description changes based on `B_MENTAL_HERB`.

**Comment by AsparagusEduardo** [comment]
We agreed that we would keep the data of all FRLG items. Please add the rest.

**Review by AsparagusEduardo — APPROVED** [review]

**Comment by AsparagusEduardo** [comment]
Let's hold on the merge of until #1731 's conflicts are fixed.

**Pull request merged**